hyperdome/client/threads.py

```
@@ -225,14 +225,19 @@ def __init__(self, session: requests.Session, server: Server, uid: str):
     @QtCore.pyqtSlot()
     def run(self):
         try:
-            new_messages = get_messages(self.server, self.session, self.uid)
-            self.signals.success.emit(new_messages)
+            message_response = get_messages(self.server, self.session, self.uid)
+            if message_response["chat_status"] == "CHAT_ACTIVE":
+                self.signals.success.emit(message_response["messages"])
+            elif message_response["chat_status"] == "CHAT_OVER":
+                self.signals.error.emit("chat ended")
         except requests.HTTPError:
             self.signals.error.emit("Counselor not in chat")
         except requests.RequestException:
             self.signals.error.emit("Error in get messages request")
         except MethodNotAllowed:
             self.signals.error.emit("not allowed")
+        except KeyError:
+            self.signals.error.emit("API error")
```

I hate swallowed exceptions
At least log it ;~;

---

soon™️ 
it's only here so I know what exceptions we're catching when I rewrite this whole thing and implement logging.

The current logging system is literally a wrapped print statement which I find unacceptable.